ouroboros-consensus/src/Ouroboros/Consensus/Node/ExitFailure.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Ouroboros.Consensus.Node.ExitFailure
+  ( -- * Pre-defined ExitFailurse
+    ExitFailure
+  , defaultExitFailure
+  , configurationError
+  , noNetwork
+  , restartWithRecovery
+  , wrongDatabase
+  , diskFull
+  , insufficientPermissions
+    -- * Get the ExitFailure of an Exception
+  , toExitFailure
+  ) where
+
+import           Control.Exception (SomeException, fromException)
+
+import           Control.Monad.Class.MonadAsync (ExceptionInLinkedThread (..))
+
+import           Ouroboros.Storage.ChainDB.API (ChainDbFailure (..))
+import           Ouroboros.Storage.FS.API.Types (FsError (..), FsErrorType (..))
+import           Ouroboros.Storage.ImmutableDB.Types (ImmutableDBError)
+import qualified Ouroboros.Storage.ImmutableDB.Types as ImmDB
+import           Ouroboros.Storage.VolatileDB.Types (VolatileDBError)
+import qualified Ouroboros.Storage.VolatileDB.Types as VolDB
+
+import           Ouroboros.Consensus.Node.DbMarker (DbMarkerError)
+import           Ouroboros.Consensus.Node.ProtocolInfo.Byron
+                     (PBftLeaderCredentialsError)
+
+{-------------------------------------------------------------------------------
+  Pre-defined ExitFailures
+-------------------------------------------------------------------------------}
+
+type ExitFailure = Int
+
+-- | Something went wrong, just restart the node.
+defaultExitFailure :: ExitFailure
+defaultExitFailure = 1
+
+-- | Something is wrong with the node configuration, the user should check it.
+--
+-- For example, for PBFT, it could be that the block signing key and the
+-- delegation certificate do not match.
+configurationError :: ExitFailure
+configurationError = 2
+
+-- | There is a problem with the network connection, the user should
+-- investigate.
+--
+-- TODO We're not yet returning this.
```

Comment:
@coot @karknu Which exception would the network layer throw when there is a network connection error?

Reply:
We could have internal "impossible" errors, but none expected due to a bad network connection. Those would not be fatal for the node.

Reply:
My thinking is that when the node is started without a working internet connection, it should be reported to the wallet so that it can give some useful instructions to the user instead of some cryptic (to most non-developers) error message. Would the lack of a working internet connection result in a `MuxError` or would it just time out and retry continuously?

How do you suggest we report such issues to the wallet?

Maybe it's best to remove that case from this PR for now and let you guys come up with a better way to report network connectivity issues?

Reply:
Can't we leave the error code and decide later on how to actually detect this?

Reply:
Good idea. This means we can merge as is.